src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<R>(items: R[], key: keyof R): GroupsMap<R> {
+  const groups: GroupsMap<R> = {};
+
+  items.forEach((item: R) => {
```

Comment:
```suggestion
  items.forEach((item) => {
```

it's redundand because TS already knows, that you items type - is array of "R" - so item type - is R